contact.go

```diff
@@ -78,3 +87,23 @@ func existsInStringSlice(set []string, find string) bool {
 	}
 	return false
 }
+
+func buildContact(publicKey *ecdsa.PublicKey) (*Contact, error) {
+	address := strings.ToLower(crypto.PubkeyToAddress(*publicKey).Hex())
+
+	id := fmt.Sprintf("0x%s", hex.EncodeToString(crypto.FromECDSAPub(publicKey)))
```

Comment:
```suggestion
	id := "0x" + hex.EncodeToString(crypto.FromECDSAPub(publicKey))
```

would avoid using reflection. Not important if you think the current version is more clear.